edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeOperator.cs

```diff
@@ -97,6 +103,23 @@ public class EdgeOperator : IKubernetesOperator, IRuntimeInfoProvider
 
                 ContractResolver = new CamelCasePropertyNamesContractResolver()
             };
+
+            if (!string.IsNullOrWhiteSpace(persistentVolumeName))
+            {
+                this.persistentVolumeName = KubeUtils.SanitizeK8sValue(persistentVolumeName);
```

Comment:
We may not want to sanitize the PV name (or the SC name below). If this is invalid then subsequent resource creation requests will fail. I don't mind having a validation to check that the names are valid though.

Reply:
Added a check during creation for this.